### PR TITLE
KAFKA-8199: Fix ClassCastException when trying to groupBy after suppress

### PR DESCRIPTION
**Pull request by ouertani**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

Fixes #8199 ClassCastException when trying to groupBy after suppress

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

https://github.com/apache/kafka/blob/da53141d931342563f5bbcf6b4846671d59e3033/streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java#L451
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ X] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@ouertani could you rebase this PR?

@bbejeck @vvcephei could you take a look?

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            @Override
+            public String[] storeNames() {
+                return new String[]{storeName};
+            }
+        };
+    }
+
+    @Override
+    public void enableSendingOldValues() {
+    }
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java:53]: maybe throw an unsupported operation exception here?

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public Processor<K, Change<V>> get() {
+        return suppressProcessor;
+    }
+    private class KTableSuppressValueGetter implements KTableValueGetter<K, V> {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java:60]: nit: missing empty line

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java [hunk 3]
```diff
@@ -401,14 +401,14 @@ public String queryableStoreName() {
...
 
-        final ProcessorSupplier<K, Change<V>> suppressionSupplier =
-            () -> new KTableSuppressProcessor<>(
+        final KTableSuppressProcessor<K, V> kTableSuppressProcessor = new KTableSuppressProcessor<>(
                 suppressedInternal,
                 storeName,
                 keySerde,
                 valSerde == null ? null : new FullChangeSerde<>(valSerde)
-            );
+        );
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java:407]: nit: reduce the indention of the passed in parameters by 4, too.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java [hunk 4]
```diff
@@ -445,6 +448,20 @@ public void suppressShouldShutDownWhenOverByteCapacity() {
             assertThat(e.getMessage(), containsString("buffer exceeded its max capacity"));
         }
     }
+    @Test
+    public void groupByAfterSuppresShouldRuns() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        final Properties properties = new Properties();
+
+        builder.<String, String>stream("topic")
+                .groupByKey().windowedBy(TimeWindows.of(Duration.ofSeconds(30))).count()
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java]: nit: move `windowBy()` to it's own line -- also `count()` (similar below)

**Comment by ouertani** [comment]
> @ouertani could you rebase this PR?
> 
> @bbejeck @vvcephei could you take a look?

@guozhangwang rebased.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java (hunk 1 above)
Comment (ouertani) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java:53]: add  throw new UnsupportedOperationException("enableSendingOldValues is not supported by KTableSuppressSupplier");

**Comment by bbejeck** [comment]
Hi @ouertani looks like there are some extra changes not related to this PR. Can you try and do a force push of the rebased branch?

**Comment by vvcephei** [comment]
Looking at the commits, it looks like it might have been a merge. The thing to do is `rebase -i trunk`, and then remove all the commits that aren't for this change.

**Comment by ouertani** [comment]
@bbejeck @vvcephei supersedes https://github.com/apache/kafka/pull/6684

**Pull request closed without merging**